docs/decisions/0049-agents-assistantsV2.md

```diff
@@ -0,0 +1,157 @@
+
+This includes optional _execution_settings_ for defining default _run_ behavior.  Since these execution settings are not part of the remote assistant definition, they are persisted in the assistant metadata for when an existing agent is retrieved.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-definition.png"  style="width: 360pt;"></kbd>
+</p>
+
+
+#### Assistant Invocation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), settings that only apply to a discrete run may be specified.  These settings are defined as `OpenAIAssistantInvocationSettings`.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-invocationsettings.png" style="width: 220pt;"></kbd>
+</p>
+
+
+#### Thread Creation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), a thread must be explicitly managed.  When doing so, thread specific settings may be specified.  These settings are defined as `OpenAIThreadCreationSettings`.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-threadcreationsettings.png" style="width: 132pt;"></kbd>
+</p>
+
+
+#### Service Configuration
+
+The `OpenAIServiceConfiguration` defines how to connect to a specific remote service, whether it be OpenAI, Azure, or proxy.  This eliminates the need to define multiple overloads for each call site that results in a connection to the remote API service (i.e. create a _client)_.
+
+> Note: This was previously named `OpenAIAssistantConfiguration`, but is not necessarily assistant specific.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-serviceconfig.png"  style="width: 360pt;"></kbd>
+</p>
+
+
+### Agent Implementation
+
+The `OpenAIAssistant` agent is roughly equivalent to its V1 form save for:
+
+- Supports settings for _assistant_, _thread_, and _run_
+- Agent definition shifts to `Definition` property
+
+<p align="center">
+<kbd><img src="diagrams/assistant-agent.png"  style="width: 720pt;"></kbd>
+</p>
+
+The following table describes the purpose of diagramed methods on the `OpenAIAssistantAgent`.
+
+|Method Name|Description|
+---|---
+**Create**|Create a new assistant agent
+**ListDefinitions**|List existing assistant definitions
+**Retrieve**|Retrieve an existing assistant
+**CreateThread**|Create an assistant thread
+**DeleteThread**|Delete an assistant thread
+**AddChatMessage**|Add a message to an assistant thread
+**GetThreadMessages**|Retrieve all messages from an assistant thread
+**Delete**|Delete the assistant agent's definition (puts agent into a terminal state)
+**Invoke**|Invoke the assistant agent (no chat)
+**GetChannelKeys**|Inherited from `Agent`
+**CreateChannel**|Inherited from `Agent`
+
+
+### Vector Store Support
+
+_Vector Store_ support is required in order to enable usage of the `file-search` tool.  
+
+In alignment with V2 treaming of the `FileClient`, the caller may also directly target `VectorStoreClient` from the _OpenAI SDK_.
+
+
+### Class Inventory
```

Comment:
All of the classes mentioned here feels like a lot of different config classes to understand and manage -- there is the definition, the execution settings, the invocation settings, and the service configuration. It feels a bit overwhelming, at first glance.

Reply:
These should all map directly to the core API:

https://platform.openai.com/docs/api-reference/assistants

I'm just mirroring the V2 behavior.  Open to suggestions if you have something in mind.

Reply:
Definition => Assistant
Thread => Thread
Execution and Invocation => Run
Service => Access